account/identity.go

```diff
@@ -138,6 +142,24 @@ func (m *GormIdentityRepository) Load(ctx context.Context, id uuid.UUID) (*Ident
 	return &native, errs.WithStack(err)
 }
 
+// Exists returns true|false where an object exists with an identifier
+func (m *GormIdentityRepository) Exists(ctx context.Context, id uuid.UUID) (bool, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "identity", "exists"}, time.Now())
+	var exists bool
+	query := fmt.Sprintf(`
+		SELECT EXISTS (
+			SELECT 1 FROM %[1]s
+			WHERE
+				id=$1
+				AND deleted_at IS NULL
+		)`, m.TableName())
```

Comment:
I wonder if the package that provides the `Exister` interface can also implement the logic for finding an object and checking if it exist. Your `Exists` implementations almost always look the same and you might as well make a call to a function like this:

```go
func (m *GormIdentityRepository) Exists(ctx context.Context, id uuid.UUID) (bool, error) {
  return repository.Exists(ctx, m.db, m.TableName(), id, "identity")
}
```

which calls

```go
package repository

// ...

func Exists(ctx context.Context, db *gorm.DB, tableName string, entityID uuid.UUID, descriptiveName string) (bool, error) {
  defer goa.MeasureSince([]string{"goa", "db", descriptiveName, "exists"}, time.Now())
  var exists bool
  query := fmt.Sprintf(`
 		SELECT EXISTS (
 			SELECT 1 FROM %[1]s
 			WHERE
 				id=$1
 				AND deleted_at IS NULL
 		)`, tableName)
  if err := db.CommonDB().QueryRow(query, id).Scan(&exists); err != nil {
    return false, errs.Wrapf(err, "failed to check if an %s exists with this id %v", id, descriptiveName)
  }
  return exists, nil
}
```